core/commands/storage/upload/upload/init.go

```diff
@@ -235,6 +236,62 @@ the shard and replies back to client for the next challenge step.`,
 				if err != nil {
 					return err
 				}
+				go func() {
+					if question == nil {
+						return
+					}
+					ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+					defer cancel()
+
+					fileHash, err := cidlib.Parse(req.Arguments[1])
+					if err != nil {
+						return
+					}
+					shardHash, err := cidlib.Parse(question.Question.ShardHash)
+					if err != nil {
+						return
+					}
+					sc, err := challenge.NewStorageChallengeResponse(ctx, ctxParams.N, ctxParams.Api,
+						fileHash, shardHash, "", false, 0)
+					if err != nil {
+						return
+					}
+					err = sc.SolveChallenge(int(question.Question.ChunkIndex), question.Question.Nonce)
+					if err != nil {
+						return
+					}
+					resp := &guardpb.ResponseChallengeQuestion{
+						Answer: &guardpb.ChallengeQuestion{
+							ShardHash:    question.Question.ShardHash,
+							HostPid:      question.Question.HostPid,
+							ChunkIndex:   int32(sc.CIndex),
+							Nonce:        sc.Nonce,
+							ExpectAnswer: sc.Hash,
+						},
+						HostPid:     question.Question.HostPid,
+						ResolveTime: time.Now(),
+					}
+					privKey, err := ctxParams.Cfg.Identity.DecodePrivateKey("")
+					if err != nil {
+						return
+					}
+					sig, err := crypto.Sign(privKey, resp)
+					if err != nil {
+						return
+					}
+					resp.Signature = sig
+					err = grpc.GuardClient(ctxParams.Cfg.Services.GuardDomain).WithContext(ctx,
+						func(ctx context.Context, client guardpb.GuardServiceClient) error {
+							_, err := client.ResponseChallenge(ctx, resp)
+							if err != nil {
+								return err
+							}
+							return nil
+						})
+					if err != nil {
+						return
+					}
+				}()
```

Comment:
shard is not complete until the above is done, so it should not launch a new goroutine